features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/ImageButtonMapper.kt

```diff
@@ -27,7 +27,7 @@ internal class ImageButtonMapper(
         view: ImageButton,
         mappingContext: MappingContext
     ): List<MobileSegment.Wireframe.ImageWireframe> {
-        val drawable = view.drawable
+        val drawable = view.drawable?.constantState?.newDrawable()
```

Comment:
Why this call is needed? As I see, `constantState` can return `null`, won't we loose drawables from some drawables? Also `newDrawable` is saying the following:

```
Note that using this means the density-dependent drawables (like bitmaps) will not be able to update their target density correctly. One should use {@link #newDrawable(Resources)} instead to provide a resource.
```
Shouldn't we use `newDrawable(Resources)` instead?

Reply:
Needed because drawing to canvas using the original drawable seems to mutate the view in the app - this became quite visible with downscaling

Reply:
I believe that for __certain__ drawables (which have the state) call to `draw` can indeed modify their state (drawable itself don't have a link to the view and the canvas passed to the `draw` method is owned by us, so it is just about the state of drawable). I don't think we need this allocation for the static (read __stateless__) drawables, so I guess this is the field where we can optimize later.

Let's create a task for optimization to look deeper into this topic for later and keep it like that for now maybe?